### PR TITLE
Magiclysm monsters I

### PR DESCRIPTION
**Pull request by arijust**
#### Summary

```SUMMARY: Mods "Magiclysm - New Monsters I"```
#### Purpose of change

Introduces three monsters:
Stirge (magical mix of mosquito and bat, with special attack "blood suck")
Shrieker (magic fungus, which makes large noise when anything gets too close to it)
Lemure (infernal larva - which after sometime will evolve into imps/fiends etc.)

Stirge could be added to swamps, shriekers to underground dungeons and lemures to deep underground temples. 
#### Additional context

Stirge still needs to get its Blood Suck special attack and balance checks if any, before its ready for merge.
Few kinds of imps for lemure evolution will done at other PRs.

### CODE REVIEW

File: data/mods/Magiclysm/monsters/monsters.json [hunk 1]
```diff
@@ -242,5 +242,92 @@
...
+    "vision_night": 20,
+    "harvest": "stirge_death",
+    "anger_triggers": [ "PLAYER_CLOSE" ],
+    "death_function": [ "NORMAL" ],
+    "flags": [ "SEES", "SMELLS", "HEARS", "FLIES", "HIT_AND_RUN", "PATH_AVOID_FIRE", "SWARMS" ]
+  },
+  {
+    "id": "mon_shrieker",
+    "type": "MONSTER",
+    "name": "shrikier",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json]:
```suggestion
    "name": "shrieker",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 2]
```diff
@@ -242,5 +242,92 @@
...
+    "melee_dice": 1,
+    "melee_dice_sides": 4,
+    "melee_cut": 1,
+    "dodge": 2,
+    "vision_day": 40,
+    "vision_night": 40,
+    "harvest": "exempt",
+    "anger_triggers": [ "PLAYER_CLOSE" ],
+    "death_function": [ "MELT" ],
+    "flags": [ "SEES", "SMELLS", "HEARS", "STUMBLES", "PATH_AVOID_FIRE" ]
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/monsters/monsters.json]: No regeneration factor for lemure?
image
https://forgottenrealms.fandom.com/wiki/Lemure

Comment (foulman) [reply data/mods/Magiclysm/monsters/monsters.json]: Why are you so evil?

**Comment by KorGgenT** [comment]
you might be able to emulate a blood suck attack with spells, by making a spell that does damage to an opponent and heals self as a side effect.

File: data/mods/Magiclysm/monsters/monsters.json (hunk 2 above)
Comment (LaVeyanFiend) [reply data/mods/Magiclysm/monsters/monsters.json]: 😈

Comment (arijust) [reply data/mods/Magiclysm/monsters/monsters.json]: I will add Regenerate 1 to it, but revival ability is out of question at current time.
Mainly from lack of holy element, weapons or water.

Comment (KorGgenT) [reply data/mods/Magiclysm/monsters/monsters.json]: i don't have a problem with giving it the revives flag. i don't want to add an extra element to check whether or not the revival is possible, so butchery or pulping should be fine to work close enough

**Comment by arijust** [comment]
>making a spell that does damage to an opponent and heals self as a side effect.
Seems simple enough. But into which file it should go?

**Comment by KorGgenT** [comment]
you can make a new file in spells for monster spells.

File: data/mods/Magiclysm/monsters/monsters.json (hunk 2 above)
Comment (arijust) [reply data/mods/Magiclysm/monsters/monsters.json]: Ok. I will add revives flag and change its death function into normal one, then.

**Comment by arijust** [comment]
Almost done. Now only to add harvest groups and some materials for these.
Proboscis for stirge(maybe for arrows/javelins or with other tools for liquid related work), scream mush for shrieker(to make potion called "scream in bottle") , but I don't know what to add for lemure.

File: data/mods/Magiclysm/items/comestibles.json [hunk 3]
```diff
@@ -60,5 +60,14 @@
     "name_plural": "dragon meat",
     "description": "The meat from a slain dragon.  It does not seem to respond to cooking the same way as other meat, making it obvious that it still retains some magical properties.",
     "parasites": 0
+  },
+  {
+    "id": "scream_mush",
+    "type": "COMESTIBLE",
+    "copy-from": "mushroom",
+    "name": "scream mushroom",
+    "name_plural": "scream mushrooms",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/comestibles.json]:
```suggestion
```
by default, plural adds an "s"

File: data/mods/Magiclysm/items/comestibles.json [hunk 4]
```diff
@@ -60,5 +60,14 @@
...
+    "description": "The mushrooms harvested from dead shrieker. Could be used in potions.",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/comestibles.json]:
```suggestion
    "description": "The mushrooms harvested from a dead shrieker.  Could be used in potions.",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 5]
```diff
@@ -182,5 +182,20 @@
...
     "phase": "liquid",
     "freezing_point": -100
+  },
+  {
+    "id": "stirge_proboscis",
+    "type": "GENERIC",
+    "copy-from": "wasp sting",
+    "symbol": ",",
+    "color": "red",
+    "name": "stirge proboscis",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/alchemy_items.json]: you'll need to add a plural form. "stirge proboscises" is adequate.

**Comment by arijust** [comment]
I think it should be good to go.
(Except that mess in commit history, but that way beyond my git knowledge right now.)

**Comment by KorGgenT** [comment]
all right, mark it ready for review and i'll review it soon.

**Comment by KorGgenT** [comment]
image

**Comment by KorGgenT** [comment]
here is the cata json linting tool with the rules we use for our json.
http://dev.narc.ro/cataclysm/format.html

**Comment by arijust** [comment]
I used it already, but it's still shows up error.

**Comment by KorGgenT** [comment]
if it shows a syntax error, that means you have invalid json, which you can use this to help find: https://jsonlint.com/

**Comment by arijust** [comment]
I found mistake in my JSON at monsterspells.json. It was case of incorrect spacing.

**Comment by KorGgenT** [comment]
image
please refer to[ this example of using a spell as a monster attack](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/Magiclysm/monsters/demon_spider.json#L172)

**Comment by KorGgenT** [comment]
image
i think this might be because copy-from works for id, not for name

**Comment by arijust** [comment]
Closing this.
I am unable to fix my repo without nuking it.
Feel free to copy these monsters into your personal pull requests.

**Pull request closed without merging**